packages/nest/src/feature.interceptor.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable, NestInterceptor, ExecutionContext, CallHandler } from '@nestjs/common';
+import { Observable } from 'rxjs';
+import { AsyncLocalStorage } from 'async_hooks';
+import { EvaluationContext } from '@openfeature/core';
+
+export const asyncLocalStorage = new AsyncLocalStorage<EvaluationContext>();
+
+@Injectable()
+export class OpenFeatureContextInterceptor implements NestInterceptor {
+  intercept(context: ExecutionContext, next: CallHandler): Observable<any> {
+
+    if (context.getType() === 'http') {
+      const req = context.switchToHttp().getRequest();
+      const targetingKey = req.get('x-user-id');
+      if (targetingKey) {
+
+        return asyncLocalStorage.run({targetingKey}, () => next.handle());
+      }
```

Comment:
Here we would have go get a mechanism choose what is injected as context.

Reply:
I've been thinking about using a context key -> header mapping on instantiation for this.
Something like:

```ts
new Interceptor({targetKey: 'x-user-id', tenant: 'x-tenat-id'});
```

Reply:
This would be good, but e.g. other request info like the IP address could also be needed.

Reply:
In general, I think a functional approach would be more flexible. Instead of a key/value config, you could accept a function that takes a request, and run that... then users can extract whatever they want.

Reply:
This would be something like the `contextFactory` that we have in the current draft.